### PR TITLE
Mvp bots composite

### PR DESCRIPTION
**Pull request by frUSIc**
Added method to determine relative ball distance and bearing angle from the robot.
Converted simple bodies to composite bodies to support truly independent motors.

### CODE REVIEW

File: scripts/soccer.js [hunk 1]
```diff
@@ -121,65 +122,131 @@
...
+    
     class Robot {
         /**
          * Abstract class Robot containing all explanations and methods to be overwritten
+         * @param {team} team is the team's goal colour, yellow or blue. TODO: Currently blue team faces up, Yellow is down
          * @param {x} x is the x position of the robot
          * @param {y} y is the y position of the robot
          */
-        constructor(x, y) {
+        constructor(team) {
```
Comment (kcnotes) [inline scripts/soccer.js:133]: x,y is better kept here, since we use it for both Dual and Uni

File: scripts/soccer.js [hunk 2]
```diff
@@ -121,65 +122,131 @@
...
+            }
+
+            let res = angle;
+            // If rotated rightwards
+            if (angle > 180) {
+                res = -1 * (360-angle);
+            // If rotated leftwards
+            } else if (angle < -180){
+                res = angle + 360;
+            }
```
Comment (kcnotes) [inline scripts/soccer.js:225]:
```javascript
        getBearing(){
            let angle = this.getAngle() % (2*Math.PI);

            // Compensate for yellow facing other way to blue
            if (this.team == 'blue'){
                angle += Math.PI;
            }
            angle = (angle + 2*Math.PI) % (2*Math.PI);

            if (angle > Math.PI) {
                angle -= 2*Math.PI;
            }
            return angle;
        }
```

File: scripts/soccer.js [hunk 3]
```diff
@@ -121,65 +122,131 @@
...
+
+            // Limit angle within -180 to 180 deg
+            if (angle > 180){
+                angle = -1 * (360-angle);
+            } else if (angle < -180){
+                angle += 360;
+            }
+
+            return {distance: distance, angle: angle*degToRad};
+        }
```
Comment (kcnotes) [inline scripts/soccer.js:188]:
```javascript
        getBallPosition(ball){
            let ballPos = {x: ball.position.x, y: ball.position.y},
                robPos = this.getPos(),
                delta = {x: ballPos.x - robPos.x, y: -1 * (ballPos.y - robPos.y)},
                distance = Math.sqrt( Math.pow(delta.x,2) + Math.pow(delta.y,2) ),
                ballBearing = Vector.angle(robPos,ball.position)*radToDeg;
            ballBearing += 270;
            // Adjust based on colour
            if (this.team == 'blue') {
                ballBearing += 180;
            }
            // Adjust based on robot heading
            ballBearing -= this.getBearing() * radToDeg;
            
            // Wrap between -180 and 180
            ballBearing = (ballBearing + 360) % (360);
            if (ballBearing > 180) {
                ballBearing -= 360;
            }

            return {distance: distance, angle: ballBearing*degToRad};
        }
```

Comment (frUSIc) [reply scripts/soccer.js:188]: Changed

File: scripts/soccer.js (hunk 2 above)
Comment (frUSIc) [reply scripts/soccer.js:225]: Changed

**Pull request merged**